### DISCUSSION TITLE
What is the value in creating a design system or component library for Preact?

### DISCUSSION THREAD

**Discussion by mistyharsh** [preactjs/preact#2884] [General]
I have been using Preact for quite some time and have built components on top of it. I have happily used many React components using `preact/compact` and it works flawlessly except occasionally missing TypeScript type definitions for external components.

Given this scenario that you can use existing React based component system in Preact applications, does it make sense to build a component library exclusively for Preact from ground up?

Are there any technical merits to do this and help Preact ecosystem other than the business reasons for doing so? If yes, what would be those? Also, how can we set differentiator for this proposition compared to other React based libraries?

**Comment by JoviDeCroock** [comment]
I think the main thing to consider is that a component library is a huge maintenance burden, hence why I personally love things like TailwindCSS, Tailwind removes the maintenance burden of a full-fledged component library quite a bit. You'll have to maintain the traditional components you often use like `Button`, ... but the main used css, ... will all be incorporated in the helpers they provide which starts to come close to a design-system that is framework-agnostic since it's css.

That being said a Preact-centric component library should carry over the value proposition that Preact brings to the table, we pride ourselves with performance and size, this could be a differentiating factor.

**Reply by mistyharsh** [reply]
Thanks for the reply. That nicely sums up the expectations that can we set upfront while planning for the work. We already have an internally built-in component library used in production which we intend to open-source.
